dds/DCPS/XTypes/TypeLookupService.cpp

```diff
@@ -789,7 +789,7 @@ void TypeLookupService::complete_to_dynamic_i(DynamicTypeImpl* dt,
       type_identifier_to_dynamic(cto.union_type.discriminator.common.type_id, guid);
     td->discriminator_type(disc_type);
     DDS::MemberDescriptor_var disc_md = new MemberDescriptorImpl();
-    disc_md->name("_d");
+    disc_md->name("discriminator");
```

Comment:
`_d` is a protected IDL name, but `discriminator` isn't, so could someone have a union branch named `discriminator` that conflicts with this? (Even if that's unlikely)

Reply:
There doesn't seem to be a compiler error for breaking the rule.

Reply:
That's correct.  I'm OK with it since it's such a corner case and one could actually use the type as long as operations like get_member_by_name aren't needed.